### PR TITLE
[MAT-17] Init Nomad. Test Mate via Mate.

### PR DESCRIPTION
**Pull request by gollariel**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,9 +2,9 @@ FROM golang:1.10.3-alpine as builder
 
 ADD . /go/src/github.com/patch-notes/mate
 WORKDIR /go/src/github.com/patch-notes/mate
-RUN apk update; apk add build-base upx \
- && make build \ 
- && upx mate -o /mate
+RUN apk update; apk add build-base upx git \
```
Comment (thepeak99) [inline Dockerfile]: why do you include git here? I don't see it used anywhere else... I understand maybe it is for the golang vendors but they're in the repo...

File: Makefile [hunk 2]
```diff
@@ -2,10 +2,17 @@ ALL: format build
...
 
-build:
+build_mate:
 	go build $(LDFLAGS) -o build/mate cmd/mate/main.go
 
+docker_build:
+	docker build -t mate-core ./
+
+docker_test:
+	docker run -v /var/run/docker.sock:/var/run/docker.sock -e MATE_TRIGGER=github.push --rm -it -v $(CURRENT_PATH):/work -v $(CURRENT_PATH)/environment:/env mate-core
```
Comment (thepeak99) [inline Makefile]: To make it more "correct" we could create another trigger, something like `mate.build`, and then the `github.push` trigger would call it. So here we would call `mate.build` and not `github.push`

File: cli/actions.go [hunk 3]
```diff
@@ -138,6 +138,11 @@ func getActionByType(actionRaw map[string]interface{}) (action Action, err error
 
 func runActionContainer(ctx context.Context, mate Mate, image string, environment EnvironmentVars, volumes Volumes) (err error) {
 	envFile := mate.EnvironmentFile()
+	err = envFile.Load()
```
Comment (thepeak99) [inline cli/actions.go]: this addresses the bug that I filled a couple days ago ^^

File: dockertools/docker.go [hunk 4]
```diff
@@ -90,11 +92,36 @@ func (d *dockerToolClient) RunContainer(ctx context.Context, image string, env [
...
+		if closeErr := reader.Close(); closeErr != nil {
+			err = closeErr
+		}
+	}()
+
+	// Wrap reader for cut header
+	_, err = io.Copy(os.Stdout, NewLogReader(reader))
+	if err != nil {
+		return err
+	}
```
Comment (thepeak99) [inline dockertools/docker.go]: nice, now we have stdout! :)

Comment (thepeak99) [reply dockertools/docker.go]: uhm but we're writing everything now to stdout, maybe in the future we want to make a difference, and write stdout to stdout and stderr to stderr.

File: extras/containers/mate-golang-tests/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM golang:1.10-stretch
```
Comment (thepeak99) [inline extras/containers/mate-golang-tests/Dockerfile]: why dont you create a `Dockerfile.test` file in the root of the project?

File: Dockerfile (hunk 1 above)
Comment (gollariel) [reply Dockerfile]: This is required to execute "make build_mate". We are using it for getting a version

Comment (thepeak99) [reply Dockerfile]: ah ok, understood!

Comment (thepeak99) [reply Dockerfile]: but I dont see it in build mate haha
```
build_mate:
 	go build $(LDFLAGS) -o build/mate cmd/mate/main.go
```

File: extras/containers/mate-golang-tests/Dockerfile (hunk 5 above)
Comment (gollariel) [reply extras/containers/mate-golang-tests/Dockerfile]: I do not like to have a lot of different Dockerfiles in one directory... If it makes sense, I can move it in the root path

File: Dockerfile (hunk 1 above)
Comment (thepeak99) [reply Dockerfile]: ah wait i saw it, nothing, forget haha

File: dockertools/docker.go (hunk 4 above)
Comment (gollariel) [reply dockertools/docker.go]: Sure, I added a comment about header different in LogReader... we can just make a special function in LogReader, for copy different messages in the correct place...

File: Makefile (hunk 2 above)
Comment (gollariel) [reply Makefile]: Agree. Let me rename it

File: cli/actions.go (hunk 3 above)
Comment (gollariel) [reply cli/actions.go]: Oh... cool :+1:

File: extras/containers/mate-golang-tests/Dockerfile (hunk 5 above)
Comment (thepeak99) [reply extras/containers/mate-golang-tests/Dockerfile]: hehe it is fine to put several dockerfiles in one directory, we already have `Dockerfile` and `Dockerfile.dev`, it is perfectly fine to have `Dockerfile.test`, and I want to encourage that with Mate ;)

File: dockertools/docker.go (hunk 4 above)
Comment (thepeak99) [reply dockertools/docker.go]: yep, for now it is okay, let's create a ticket for that and study it later :)

**Comment by gollariel** [comment]
Updated

**Pull request merged**